Add image-text-to-text and edit image-to-text task pages (#553)
huggingface/huggingface.js: packages/tasks/src/tasks/image-text-to-text/about.md

## Use Cases

> ### Visual Question Answering

Reviewer: I would avoid to put examples of things that are covered in other task pages to avoid confusion. The 3 examples now are already covered. E.g.

- VQA https://huggingface.co/tasks/visual-question-answering
- Doc QA https://huggingface.co/tasks/document-question-answering
- Captioning https://huggingface.co/tasks/image-to-text

Author: I think I disagree here. one could segment humans with both image segmentation and mask detection for instance (except for zero shot part). some hugging face tasks are similar. some models are very generalistic that they can handle these tasks altogether and the paradigm is shifting more towards there anyway.

these models are similar either way, if I were an MLE who had to do captioning I'd like to try both VLMs (BTW which perform better IMO) and direct captioning models (image-to-text) which has sole purpose of captioning. I wouldn't like to keep this information from the user.

Author: I think it would be too repetitive. See here for instance the one in data.ts will be at the top, and then right after use case one repeats it. I wouldn't like to add it there
<img width="617" alt="Screenshot 2024-03-25 at 20 49 41" src="https://github.com/huggingface/huggingface.js/assets/53175384/06ee6f3c-0cd0-4303-ab00-3d7f844a52ca">

Reviewer: My main concern is that now we have two tasks that cover the same thing, so it could end up confusing users. As an example, imagine if https://huggingface.co/tasks/text-generation use cases were Translation and Summarization, which are also their own separate tasks.

I wonder if we can use here specific applications rather than sub-tasks covered here (thinking of https://huggingface.co/tasks/text-generation#use-cases as a nice example)

Author: yes I agree, but I think as a specific/solid use case there's not a lot (they'd be way too specific)

Author: an organic way for me to infer use cases for tasks were checking out example inputs to many Spaces based on a specific task, and for VLMs it's mostly visual question answering for instance.

Author: I added bunch of other things

Vision language models trained on image text prompt pairs can be used for visual question answering.

## Document Question Answering and Retrieval

Vision language models trained on document text, layout and graphics can be used to extract information from documents.

### Image Captioning

Vision language models can be used to generate captions for images.

## Inference

> You can use the Transformers library's `image-text-to-text` pipeline to interact with vision language models.

Reviewer 2: ```suggestion
You can use the Transformers library's `image-text-to-text` pipeline to interact with vision-language models.
```

> You can use the 🤗 Transformers library's `image-text-to-text` pipeline to interact with vision language models.

Reviewer: ```suggestion
You can use the Transformers library's `image-text-to-text` pipeline to interact with vision language models.
```

## Use Cases

### Multimodal Dialogue

> Vision language models can be used as multimodal assistants, keeping context about the conversation and keeping the image to have multiple turn dialogues.

Reviewer: ```suggestion
Vision language models can be used as multimodal assistants, keeping context about the conversation and keeping the image to have multiple-turn dialogues.
```

### Image Recognition with Instructions

Vision language models can recognize images through descriptions. When given detailed descriptions of specific entities, it can classify the entities in an image.

### Zero-shot Object Detection, Image Segmentation and Localization

> Some vision language models can detect or segment a set of objects or describe the positions or relative positions of the objects. For example, one could prompt such a model to ask if one object is behind another. Such a model can also output bounding box coordination or segmentation masks directly in the text output, unlike the traditional models explicitly trained on only doing object detection or image segmentation.

Reviewer: ```suggestion
Some vision language models can detect or segment a set of objects or describe the positions or relative positions of the objects. For example, one could prompt such a model to ask if one object is behind another. Such a model can also output bounding box coordination or segmentation masks directly in the text output, unlike the traditional models explicitly trained on only object detection or image segmentation.
```

### Visual Question Answering

Vision language models trained on image-text pairs can be used for visual question answering.

### Document Question Answering and Retrieval

Vision language models trained on formatted documents that include text and graphics can extract information from documents. This includes OCR-free retrieval as well, where one can directly feed an image without putting the image through OCR and feeding the text.

### Image Captioning

> Vision language models can be used to generate captions for images.

Reviewer: I feel this is a bit redundant/repetitive with the heading. Maybe everything can be encompassed under zero-shot capabilities, and have a bullet point list of some example tasks

## Inference

You can use the Transformers library's `image-text-to-text` pipeline to interact with vision-language models.

```python
from transformers import pipeline

mm_pipeline = pipeline("image-text-to-text", model="microsoft/kosmos-2-patch14-224")

generated_text = mm_pipeline(images="https://huggingface.co/spaces/llava-hf/llava-4bit/resolve/main/examples/baklava.png", text="How to make this pastry?", max_new_tokens=50)
## [{'generated_text': 'How to make this pastry? 1. Preheat oven to 200 degrees Celsius.'}]
```

## Useful Resources

> 

Reviewer: Maybe https://huggingface.co/blog/idefics too

## Different Types of Vision Language Models

Vision language models come in three types:

> - **Base:** Pre-trained models that can be fine-tuned. A good example is for base models is [PaliGemma models](https://huggingface.co/models?sort=trending&search=google%2Fpaligemma-3b-pt) by Google.

Reviewer: ```suggestion
- **Base:** Pre-trained models that can be fine-tuned. A good example of base models is the [PaliGemma models family](https://huggingface.co/models?sort=trending&search=google%2Fpaligemma-3b-pt) by Google.
```

> - **Instruction:** Base models fine-tuned on instruction datasets. A good example for instruction fine-tuned models is [idefics2-8b](https://huggingface.co/HuggingFaceM4/idefics2-8b).

Reviewer: ```suggestion
- **Instruction:** Base models fine-tuned on instruction datasets. A good example of instruction fine-tuned models is [idefics2-8b](https://huggingface.co/HuggingFaceM4/idefics2-8b).
```

- **Chatty/Conversation:** Base models fine-tuned on conversation datasets. A good example for chatty models is [deepseek-vl-7b-chat](https://huggingface.co/deepseek-ai/deepseek-vl-7b-chat).

![VLM uses](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/blog/vlm/visual.jpg)

## Use Cases

### Multimodal Dialogue

Vision language models can be used as multimodal assistants, keeping context about the conversation and keeping the image to have multiple-turn dialogues.

### Zero-shot Object Detection, Image Segmentation and Localization

Some vision language models can detect or segment a set of objects or describe the positions or relative positions of the objects. For example, one could prompt such a model to ask if one object is behind another. Such a model can also output bounding box coordination or segmentation masks directly in the text output, unlike the traditional models explicitly trained on only object detection or image segmentation.

### Visual Question Answering

Vision language models trained on image-text pairs can be used for visual question answering and generating captions for images.

### Document Question Answering and Retrieval

> Documents often consist of different layouts, charts, tables, images and more. Vision language models trained on formatted documents can extract information from documents. This is an OCR-free approach, the inputs skip OCR and documents are directly fed to vision language models.

Reviewer: ```suggestion
Documents often consist of different layouts, charts, tables, images, and more. Vision language models trained on formatted documents can extract information from them. This is an OCR-free approach; the inputs skip OCR, and documents are directly fed to vision language models.
```

### Image Recognition with Instructions

Vision language models can recognize images through descriptions. When given detailed descriptions of specific entities, it can classify the entities in an image.

## Inference

You can use the Transformers library to interact with vision-language models. You can load the model like below.

```python
from transformers import LlavaNextProcessor, LlavaNextForConditionalGeneration
import torch

device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
processor = LlavaNextProcessor.from_pretrained("llava-hf/llava-v1.6-mistral-7b-hf")
model = LlavaNextForConditionalGeneration.from_pretrained(
    "llava-hf/llava-v1.6-mistral-7b-hf",
    torch_dtype=torch.float16,

>     low_cpu_mem_usage=True

Reviewer: I don't think we should put such a feature (somewhat experimental) in the task pages
